Automattic/vip
# fix(dev-env): Force IPv4 for services bound to localhost

## Description

Lando's `scanUrls()` may mark the service as down if its URL mentions `localhost` and `localhost` is resolved to `::1` instead of `127.0.0.1`.

This happens because Docker has [issues with IPv6](https://docs.docker.com/config/daemon/ipv6/); therefore, services do not bind to `::1` and requests fail:

```
what-mu 17:30:15 DEBUG ==> checking to see if http://localhost:59642 is ready. 
what-mu 17:30:15 DEBUG ==> scan failed for http://localhost:59642 code=ECONNREFUSED, message=connect ECONNREFUSED ::1:59642
what-mu 17:30:15 DEBUG ==> No response for http://localhost:59642. Setting to bad 
what-mu 17:30:15 DEBUG ==> http://localhost:59642 not currently accessible 
```

This PR updates Lando with the patched version which fixes the above bug.

As a bonus, `isEnvUp()` now checks only web services; services bound to `localhost` are explicitly ignored. This speeds up the readiness check and does not mark environment as down if a secondary service is dead.
## Steps to Test

Add `::1 localhost` to `/etc/hosts` or its local equivalent.

Create a dev environment with PMA and MailHog and start it. Without the patch, it will show the environment as down; with the patch, everything should be OK.
